### PR TITLE
Feature/cairngorms

### PR DESCRIPTION
**Pull request by shenavall73**
- Add configuration for habmos_Cairngorms and SpaceInt_Cairngorms spatial frameworks
- If the current zoom level matches the initial default zoom, and a new spatial framework is selected, pan map to centre of selected framework
- Refactor so that the names given to the JSON structures in frameworks.ts are not as used as the framework table names for constructing queries  

May need some discussion, maybe schedule a chat to talk over code?

### CODE REVIEW

File: www/src/slices/mapper/slice.ts [hunk 1]
```diff
@@ -132,3 +132,7 @@ let slice = createSlice({
 
 export let mapperReducer = slice.reducer
 export let mapperActions = slice.actions
+
+export function isCurrentZoomDefault(zoom: number) {
```
Comment (CathyJinJNCC) [inline www/src/slices/mapper/slice.ts:136]: I wonder if it's more in line with react to make this part of the state. E.g. on line 18 we have a "zoomedEnoughToShowPolygons" state so we could have some kind of "panToFramework" state?

**Review by CathyJinJNCC — APPROVED** [review]

File: www/src/slices/mapper/LeafletMap.tsx [hunk 2]
```diff
@@ -84,6 +84,9 @@ export let LeafletMap = () => {
   // redraw boundary for change of `framework`
   useEffect(() => {
     dispatch(mapperActions.selectPolygon(undefined))
+    if (isCurrentZoomDefault(state.zoom)) {
+      map.setView(frameworks[state.query.framework].defaultQuery.center, state.zoom)
```
Comment (CathyJinJNCC) [inline www/src/slices/mapper/LeafletMap.tsx:88]: Also there is a flyTo method which looks a bit more swish but the small basemap means you can't centre properly so it kind of bounces you back which is a bit jarring. I might fiddle with this for fun at some point

File: www/src/frameworks.ts [hunk 3]
```diff
@@ -55,5 +56,51 @@ export const frameworks: {[framework: string]: Framework} = {
...
+      framework: 'habmos_Cairngorms',
+      center: {lat: 57.1, lng: -3.7},
+      indexname: 'NDVI',
+      statistic: 'mean',
+      yearFrom: 2020,
+      monthFrom: 1,
+      yearTo: 2021,
+      monthTo: 12,
+    },
+    availableDates: {from: {year: 2015, month: 7}, to: {year: 2022, month: 5}},
```
[the remark this answers is not in the document]
Comment (shenavall73) [reply www/src/frameworks.ts]: Fixed

File: www/src/slices/mapper/FrameworkPanel.tsx [hunk 4]
```diff
@@ -28,7 +28,7 @@ export let FrameworkPanel = () => {
           </select>
 
           <div className="flex gap-2 items-center ml-1.5">
-            <div className="little-label-text ">{query.framework}</div>
+            <div className="little-label-text ">{frameworks[query.framework].defaultQuery.framework}</div>
```
[the remark this answers is not in the document]
Comment (shenavall73) [reply www/src/slices/mapper/FrameworkPanel.tsx:31]: Can talk through it if you'd like

File: www/src/slices/mapper/LeafletMap.tsx (hunk 2 above)
Comment (shenavall73) [reply www/src/slices/mapper/LeafletMap.tsx:88]: OK, so I take it we will commit this as-is for now

**Pull request merged**